DxDispatch/src/dxdispatch/Device.cpp

```diff
@@ -442,10 +455,13 @@ std::vector<std::byte> Device::Download(Microsoft::WRL::ComPtr<ID3D12Resource> d
     }
 
     ExecuteCommandListAndWait();
-
-    std::vector<std::byte> outputBuffer(defaultBuffer->GetDesc().Width);
+    if(defaultBuffer->GetDesc().Width > std::numeric_limits<size_t>::max())
+    {
+        throw std::invalid_argument(fmt::format("Buffer width '{}' is too large.", defaultBuffer->GetDesc().Width));
+    }
+    std::vector<std::byte> outputBuffer(static_cast<size_t>(defaultBuffer->GetDesc().Width));
     {
-        size_t dataSize = defaultBuffer->GetDesc().Width;
+        size_t dataSize = static_cast<size_t>(defaultBuffer->GetDesc().Width);
```

Comment:
`gsl::narrow` would be the cleaner option in this case, indicating it's a narrowing conversion rather than any old static cast (like the line below).